llama/model.py

```diff
@@ -204,9 +204,8 @@ def forward(
 
         xq, xk = apply_rotary_emb(xq, xk, freqs_cis=freqs_cis)
 
-        self.cache_k = self.cache_k.index_copy(1, input_indexes, xk)
-        self.cache_v = self.cache_v.index_copy(1, input_indexes, xv)
-
+        self.cache_k = self.cache_k.index_copy_(1, input_indexes, xk)
```

Comment:
`self.cache_k.index_copy_(1, input_indexes, xk)` should be enough. I think you don't need to do the assignment.

Reply:
Done.